onyxia-api/src/main/java/fr/insee/onyxia/api/openapi/OpenApiConfiguration.java

```diff
@@ -1,12 +1,72 @@
 package fr.insee.onyxia.api.openapi;
 
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
+import io.swagger.v3.oas.models.security.OAuthFlow;
+import io.swagger.v3.oas.models.security.OAuthFlows;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springdoc.core.customizers.OperationCustomizer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
-import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
-import io.swagger.v3.oas.annotations.security.SecurityScheme;
 
 @Configuration
-@SecurityScheme(type = SecuritySchemeType.HTTP, scheme = "bearer", bearerFormat = "JWT",name = "auth")
 public class OpenApiConfiguration {
 
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenApiConfiguration.class);
+
+    @Value("${keycloak.auth-server-url}")
+    public String keycloakUrl;
+
+    @Value("${keycloak.realm}")
+    public String realmName;
+
+    public final String SCHEMEKEYCLOAK = "oAuthScheme";
+
+    @Bean
+    @ConditionalOnProperty(name = "com.example.demo.auth.type", havingValue = "openidconnect", matchIfMissing = true)
+    public OpenAPI customOpenAPIKeycloak() {
+        final OpenAPI openapi = createOpenAPI();
+        openapi.components(new Components().addSecuritySchemes(SCHEMEKEYCLOAK, new SecurityScheme()
+                .type(SecurityScheme.Type.OAUTH2).in(SecurityScheme.In.HEADER).description("Authentification keycloak")
+                .flows(new OAuthFlows().authorizationCode(new OAuthFlow()
+                        .authorizationUrl(keycloakUrl + "/realms/" + realmName + "/protocol/openid-connect/auth")
+                        .tokenUrl(keycloakUrl + "/realms/" + realmName + "/protocol/openid-connect/token")
+                .refreshUrl(keycloakUrl + "/realms/" + realmName + "/protocol/openid-connect/token")))));
+        return openapi;
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "authentication.mode", havingValue = "none")

```

Comment:
This should probably be `@ConditionalOnSingleCandidate(OpenAPI.class)`.  
We want this to be the default if no specific implementation is enabled